hooks/useRailgunTx.tsx

```diff
@@ -0,0 +1,188 @@
+import { useToken } from "@/contexts/TokenContext";
+import useShieldPrivateKey from "@/hooks/useShieldPrivateKey";
+import {
+  gasEstimateForShield,
+  gasEstimateForShieldBaseToken,
+  populateShield,
+  populateShieldBaseToken,
+} from "@railgun-community/quickstart";
+import {
+  RailgunERC20AmountRecipient,
+  RailgunERC20Amount,
+  NetworkName,
+  TransactionGasDetailsSerialized,
+  EVMGasType,
+  NETWORK_CONFIG,
+  deserializeTransaction,
+} from "@railgun-community/shared-models";
+import { ethers } from "ethers";
+import { getAddress, parseUnits } from "ethers/lib/utils.js";
+import { useAccount, useSigner } from "wagmi";
+import { useProvider } from "wagmi";
+
+const useRailgunTx = () => {
+  const { data: signer } = useSigner();
+  const { address } = useAccount();
+  const { getShieldPrivateKey } = useShieldPrivateKey();
+  const network = NetworkName.EthereumGoerli;
+  const { weth } = useToken();
+  const provider = useProvider();
+
+  const shield = async (args: {
+    tokenAddress: string;
+    tokenAmount: string;
+    tokenDecimals: number;
+    recipient: string;
+  }) => {
+    if (args.tokenAddress === getAddress(`0x${"e".repeat(40)}`))
```

Comment:
Wdyt of having a `constants.ts` file, which defines all frontend constants—supported chains, ETH address, WETH addresses per network, any ABIs, etc? It's a pattern I've used in frontends and find it very helpful. If we do it, it should be a separate issue / PR

Reply:
i'm down!

Reply:
tracking in #49

Reply:
update: i just made the constants file in 1836e62